src/librustc/ty/context.rs

```diff
@@ -685,6 +697,7 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     pub fn create_and_enter<F, R>(s: &'tcx Session,
                                   local_providers: ty::maps::Providers<'tcx>,
                                   extern_providers: ty::maps::Providers<'tcx>,
+                                  mir_passes: Rc<Passes>,
```

Comment:
Oh that is annoying, I did say something to @pnkfelix along the lines of moving the MIR code in borrowck in `rustc_mir`. Unlike you I don't think we could sustain a more diffuse flow.
At least not on transformations, maybe only on analysis checks.